tools/ExampleExtractor/ExtractorAndTesterUsersGuide.md

```diff
@@ -0,0 +1,483 @@
+  - [Expected Compiler Errors](#Expected-Compiler-Errors)
+  - [Expected Compiler Warnings](#Expected-Compiler-Warnings)
+  - [Ignoring Compiler Warnings](#Ignoring-Compiler-Warnings)
+  - [Expected Runtime Output](#Expected-Runtime-Output)
+  - [Expected Exception](#Expected-Exception)
+- [Other Information](#Other-Information)
+  - [Unsafe Code](#Unsafe-Code)
+  - [Examples Containing Pseudo-Code](#Examples-Containing-Pseudo-Code)
+  - [Using Chevron-Quotes for Emphasis](#Using-Chevron-Quotes-for-Emphasis)
+  - [Tips for Creating New Testable Examples](#Tips-for-Creating-New-Testable-Examples)
+
+## Introduction
+
+The C# specification contains many source-code examples, some complete and others partial. When changing existing examples or adding new ones, it is useful to have some automatic way to extract, compile, and possibly execute them, to ensure their correctness, or to re-confirm expected warnings, errors, and/or exceptions. This is done by annotating examples, extracting them using the ExampleExtractor tool, and testing them using the ExampleTester tool.
+
+An example is annotated directly, prior to its occurrence in a spec md file, as described below in [Annotation Format](#Annotation-Format). The extraction and testing stages are part of PR processing within GitHub.
+
+## Annotation Format
+
+### General
+
+An *example_annotation* is an HTML comment having a particular structure, as shown by the ANTLR grammar below. It uses [JSON Data Interchange Syntax](https://www.ecma-international.org/publications-and-standards/standards/ecma-404/) to control the extraction and testing processes via a series of *annotation_directive*s containing JSON name/value pairs.
+
+```ANTLR
+example_annotation
+    : '<!-- Example:' '{' annotation_directive (',' annotation_directive)* '}' '-->'
+    ;
+annotation_directive
+    : template
+    | name
+    | replace_ellipsis
+    | expected_errors
+    | expected_warnings
+    | ignored_warnings
+    | expected_output
+    | expected_exception
+    ;
+```
+
+While an *example_annotation* must precede its corresponding example, it need not be immediately prior. The two can be separated by text and/or HTML comments. However, if multiple *example_annotation*s precede the same code block, the second and subsequent *example_annotation*s are ignored.
+
+Arbitrary horizontal whitespace is permitted between any two adjacent tokens.
+
+If multiple *annotation_directive*s contain the same name, the right-most directive prevails.
+
+The ordering of annotation directives is not significant.
+
+The C# spec contains two kinds of source-code examples: those within an \*Example\* and those not, as follows:
+
+````
+> *Example*:
+>
+> <!-- Example: { … } -->
+> ```csharp
+> // source code goes here
+> ```
+>
+> *end example*
+````
+
+In this case, the *example_annotation* is preceded by `> ` to match the example’s formatting. However, in the following case, it is not, as no \*Example\* is present:
```

Comment:
Out of interest, have you run across these while annotating? I didn't... the processor supports it, but maybe we don't need it.

Reply:
There are examples like this in documentation comments.